cognite/powerops/resync/v2/shop_to_assets.py

```diff
@@ -0,0 +1,370 @@
+        assets_by_xid: dict[str, GeneratorWrite | ReservoirWrite | PlantWrite | WatercourseWrite | PriceAreaWrite] = {}
+        for watercourse_dir, shop_model in self.shop_model_by_directory.items():
+            watercourse_assets = self._model_to_assets(shop_model, watercourse_dir, assets_by_xid)
+            assets_by_xid.update(watercourse_assets)
+
+        return list(assets_by_xid.values())
+
+    def _model_to_assets(
+        self,
+        shop_model: dict[str, Any],
+        watercourse_dir: str,
+        existing: dict[str, GeneratorWrite | ReservoirWrite | PlantWrite | WatercourseWrite | PriceAreaWrite],
+    ) -> dict[str, GeneratorWrite | ReservoirWrite | PlantWrite | WatercourseWrite | PriceAreaWrite]:
+        assets_by_xid: dict[str, GeneratorWrite | ReservoirWrite | PlantWrite | WatercourseWrite | PriceAreaWrite] = {}
+        try:
+            watercourse_config = self.watercourse_by_directory[watercourse_dir]
+        except KeyError as e:
+            raise ValueError(f"Watercourse directory {watercourse_dir} not watercourses.yaml file") from e
+        watercourse_external_id = f"watercourse_{watercourse_config['name']}"
+        plant_display_name_and_order = watercourse_config.get("plant_display_names_and_order", {})
+        reservoir_display_name_and_order = watercourse_config.get("reservoir_display_names_and_order", {})
+
+        generator_by_name = {}
+        for name, data in shop_model["model"]["generator"].items():
+            generator = self._to_generator(name, data)
+            if generator.external_id in existing:
+                raise ValueError(f"Generator with external id {generator.external_id} already exists")
+            generator_by_name[generator.name] = generator
+            assets_by_xid[generator.external_id] = generator
+
+        reservoir_by_name = {}
+        for name in shop_model["model"]["reservoir"].keys():
+            reservoir = self._to_reservoir(name, reservoir_display_name_and_order)
+            if reservoir.external_id in existing:
+                raise ValueError(f"Reservoir with external id {reservoir.external_id} already exists")
+            reservoir_by_name[reservoir.name] = reservoir
+            assets_by_xid[reservoir.external_id] = reservoir
+
+        plants = []
+        for name, data in shop_model["model"]["plant"].items():
+            plant = self._to_plant(
+                name,
+                data,
+                plant_display_name_and_order,
+                watercourse_external_id,
+                reservoir_by_name,
+                generator_by_name,
+                shop_model,
+            )
+            if plant.external_id in existing:
+                raise ValueError(f"Plant with external id {plant.external_id} already exists")
+            plants.append(plant)
+            assets_by_xid[plant.external_id] = plant
+
+        data = watercourse_config
```

Comment:
**suggestion:** unnecessary variable reassignment

Reply:
It is to make the lines below slightly more readable.